In [15]:
import pandas as pd
import matplotlib as plt
from pathlib import Path
import ast

pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)

In [16]:
path_app_list = Path('/home/alex/Documentos/meu-projeto/data/raw/app_list.csv')
path_steam_app_data = Path('/home/alex/Documentos/meu-projeto/data/raw/steam_app_data.csv')
path_steamspy_data = Path('/home/alex/Documentos/meu-projeto/data/raw/steamspy_data.csv')

In [17]:
df_app_list = pd.read_csv(path_app_list)
df_steam_app_data = pd.read_csv(path_steam_app_data)
df_steamspy_data = pd.read_csv(path_steamspy_data)

### Inspecting the 'df_app_list' schema

I want to check the number of columns, data type, and how many missing values we have

In [18]:
df_app_list.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29235 entries, 0 to 29234
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   appid   29235 non-null  int64 
 1   name    29230 non-null  object
dtypes: int64(1), object(1)
memory usage: 456.9+ KB


** Observations: **

- 5 null entries in 'name' columns
- 'name' column with object type

### Previewing the first and last records of 'df_app_list'

This helps indentify obvius formatting issues, inconsistent values, and irregular patterns.

In [19]:
df_app_list

appid                        name
0           10              Counter-Strike
1           20       Team Fortress Classic
2           30               Day of Defeat
3           40          Deathmatch Classic
4           50   Half-Life: Opposing Force
...        ...                         ...
29230  1065230             Room of Pandora
29231  1065570                   Cyber Gun
29232  1065650            Super Star Blast
29233  1066700  New Yankee 7: Deer Hunters
29234  1069460                   Rune Lord

[29235 rows x 2 columns]

** Observations: **

Possible parent table with appid being primary key(PK)
Switch column name "appid" to "steam_appid"


### Checking missing values in ['name']

I want to confirm whether NaNs have semantic meaning or represent missing data.

In [20]:
df_app_list[df_app_list['name'].isnull()]

appid name
481     17760  NaN
7235   396420  NaN
9553   460250  NaN
22244  806160  NaN
27324  965340  NaN

** Observations: **

- The Nans represent missing data.

** Conclusion: **

- Drop that 5 series

###

In [21]:
df_steam_app_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29235 entries, 0 to 29234
Data columns (total 39 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   type                     29086 non-null  object 
 1   name                     29234 non-null  object 
 2   steam_appid              29235 non-null  int64  
 3   required_age             29086 non-null  float64
 4   is_free                  29086 non-null  object 
 5   controller_support       5998 non-null   object 
 6   dlc                      4975 non-null   object 
 7   detailed_description     29060 non-null  object 
 8   about_the_game           29060 non-null  object 
 9   short_description        29060 non-null  object 
 10  fullgame                 0 non-null      float64
 11  supported_languages      29072 non-null  object 
 12  header_image             29086 non-null  object 
 13  website                  19252 non-null  object 
 14  pc_requirements       

So much columns with object type
Many NaN entries, analysis requiered

In [22]:

df_steam_app_data.head(50)

type                                 name  steam_appid  required_age  \
0   game                       Counter-Strike           10           0.0   
1   game                Team Fortress Classic           20           0.0   
2   game                        Day of Defeat           30           0.0   
3   game                   Deathmatch Classic           40           0.0   
4   game            Half-Life: Opposing Force           50           0.0   
5   game                             Ricochet           60           0.0   
6   game                            Half-Life           70           0.0   
7   game       Counter-Strike: Condition Zero           80           0.0   
8   game                Half-Life: Blue Shift          130           0.0   
9   game                          Half-Life 2          220           0.0   
10  game               Counter-Strike: Source          240           0.0   
11  game                    Half-Life: Source          280           0.0   
12  game                Day of Defeat: Source          300           0.0   
13  game              Half-Life 2: Deathmatch          320           0.0   
14  game              Half-Life 2: Lost Coast          340           0.0   
15  game         Half-Life Deathmatch: Source          360           0.0   
16  game             Half-Life 2: Episode One          380           0.0   
17  game                               Portal          400           0.0   
18  game             Half-Life 2: Episode Two          420           0.0   
19  game                      Team Fortress 2          440           0.0   
20  game                          Left 4 Dead          500           0.0   
21  game                        Left 4 Dead 2          550           0.0   
22  game                               Dota 2          570           0.0   
23  game                             Portal 2          620           0.0   
24  game                          Alien Swarm          630           0.0   
25  game     Counter-Strike: Global Offensive          730           0.0   
26   NaN            Half-Life: Opposing Force          852           NaN   
27  game                     Rag Doll Kung Fu         1002           0.0   
28  game        Red Orchestra: Ostfront 41-45         1200           0.0   
29  game                        Killing Floor         1250           0.0   
30  game              SiN Episodes: Emergence         1300           0.0   
31  game              SiN Episodes: Emergence         1300           0.0   
32  game              SiN Episodes: Emergence         1300           0.0   
33  game                             Darwinia         1500           0.0   
34  game                               Uplink         1510           0.0   
35  game                               DEFCON         1520           0.0   
36  game                           Multiwinia         1530           0.0   
37  game                     Dangerous Waters         1600           0.0   
38  game              Space Empires IV Deluxe         1610           0.0   
39  game     Disciples II: Rise of the Elves          1630           0.0   
40  game       Disciples II: Gallean's Return         1640           0.0   
41  game  Iron Warriors: T - 72 Tank Command          1670           0.0   
42  game                      Space Empires V         1690           0.0   
43  game                          Arx Fatalis         1700           0.0   
44  game                     Source Filmmaker         1840           0.0   
45  game                           Earth 2160         1900           0.0   
46  game              Two Worlds Epic Edition         1930           0.0   
47  game        Dark Messiah of Might & Magic         2100           0.0   
48  game                      Quake III Arena         2200           0.0   
49  game                             Quake IV         2210          18.0   

   is_free controller_support  \
0    False                NaN   
1    False                NaN   
2    False                Na

In [23]:
df_steam_app_data[df_steam_app_data['controller_support'].notnull()]

type                                         name  steam_appid  \
20     game                                  Left 4 Dead          500   
21     game                                Left 4 Dead 2          550   
23     game                                     Portal 2          620   
25     game             Counter-Strike: Global Offensive          730   
81     game                          X3: Terran Conflict         2820   
...     ...                                          ...          ...   
29144  game                                    Hexa Path      1052760   
29158  game                                   Foxyland 2      1053730   
29190  game  Geometry May. I swear it's a nice free game      1057430   
29219  game                               Peas Adventure      1062120   
29232  game                             Super Star Blast      1065650   

       required_age is_free controller_support                       dlc  \
20              0.0   False               full                       NaN   
21              0.0   False               full                       NaN   
23              0.0   False               full                  [323180]   
25              0.0    True               full                       NaN   
81              0.0   False               full  [201310, 235940, 235960]   
...             ...     ...                ...                       ...   
29144           0.0   False               full                 [1055790]   
29158           0.0   False               full                       NaN   
29190           0.0    True               full                 [1064970]   
29219           0.0   False               full                       NaN   
29232           0.0   False               full                       NaN   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         

### Observations

- Column "controler_support" have 5998 'full' values the rest is NaN.
- By context NaN mean that don't has controller suport than, replace NaN to 'not suported'

In [24]:
df_steamspy_data

appid                        name           developer  \
0           10              Counter-Strike               Valve   
1           20       Team Fortress Classic               Valve   
2           30               Day of Defeat               Valve   
3           40          Deathmatch Classic               Valve   
4           50   Half-Life: Opposing Force    Gearbox Software   
...        ...                         ...                 ...   
29230  1065230             Room of Pandora         SHEN JIAWEI   
29231  1065570                   Cyber Gun      Semyon Maximov   
29232  1065650            Super Star Blast         EntwicklerX   
29233  1066700  New Yankee 7: Deer Hunters  Yustas Game Studio   
29234  1069460                   Rune Lord    Adept Studios GD   

                  publisher  score_rank  positive  negative  userscore  \
0                     Valve         NaN    124534      3339          0   
1                     Valve         NaN      3318       633          0   
2                     Valve         NaN      3416       398          0   
3                     Valve         NaN      1273       267          0   
4                     Valve         NaN      5250       288          0   
...                     ...         ...       ...       ...        ...   
29230           SHEN JIAWEI         NaN         3         0          0   
29231      BekkerDev Studio         NaN         8         1          0   
29232           EntwicklerX         NaN         0         1          0   
29233  Alawar Entertainment         NaN         2         0          0   
29234  Alawar Entertainment         NaN         4         0          0   

                         owners  average_forever  average_2weeks  \
0      10,000,000 .. 20,000,000            17612             709   
1       5,000,000 .. 10,000,000              277              15   
2       5,000,000 .. 10,000,000              187               0   
3       5,000,000 .. 10,000,000              258               0   
4       5,000,000 .. 10,000,000              624               0   
...                         ...              ...             ...   
29230               0 .. 20,000                0               0   
29231               0 .. 20,000                0               0   
29232               0 .. 20,000                0               0   
29233               0 .. 20,000                0               0   
29234               0 .. 20,000                0               0   

       median_forever  median_2weeks  price  initialprice  discount  \
0                 317             26  999.0         999.0       0.0   
1                  62             15  499.0         499.0       0.0   
2                  34              0  499.0         499.0       0.0   
3                 184              0  499.0         499.0       0.0   
4                 415              0  499.0         499.0       0.0   
...               ...            ...    ...           ...       ...   
29230               0              0  254.0         299.0      15.0   
29231               0              0  119.0         199.0      40.0   
29232               0              0  424.0         499.0      15.0   
29233               0              0  699.0         699.0       0.0   
29234               0              0  559.0         699.0      20.0   

                                                                                                         languages  \
0               English, French, German, Italian, Spanish - Spain, Simplified Chinese, Traditional Chinese, Korean   
1      English, French, German, Italian, Spanish - Spain, Korean, Russian, Simplified Chinese, Traditional Chinese   
2                                                                English, French, German, Italian, Spanish - Spain   
3      English, French, German, Italian, Spanish - Spain, Korean, Russian, Simplified Chinese, Traditional Chinese   
4                                                                  

In [25]:
df_steamspy_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29235 entries, 0 to 29234
Data columns (total 20 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   appid            29235 non-null  int64  
 1   name             29230 non-null  object 
 2   developer        29037 non-null  object 
 3   publisher        28939 non-null  object 
 4   score_rank       58 non-null     float64
 5   positive         29235 non-null  int64  
 6   negative         29235 non-null  int64  
 7   userscore        29235 non-null  int64  
 8   owners           29235 non-null  object 
 9   average_forever  29235 non-null  int64  
 10  average_2weeks   29235 non-null  int64  
 11  median_forever   29235 non-null  int64  
 12  median_2weeks    29235 non-null  int64  
 13  price            29206 non-null  float64
 14  initialprice     29213 non-null  float64
 15  discount         29213 non-null  float64
 16  languages        29141 non-null  object 
 17  genre       

In [26]:
df_steamspy_data[df_steamspy_data['name'].isnull()]

appid name       developer       publisher  score_rank  positive  \
481     17760  NaN             NaN             NaN         NaN         0   
7235   396420  NaN             NaN             NaN         NaN        22   
9553   460250  NaN  Jeroen Wimmers  Jeroen Wimmers         NaN        44   
22244  806160  NaN    Paleno Games    Paleno Games         NaN         4   
27324  965340  NaN      2nd Studio      2nd Studio         NaN        31   

       negative  userscore            owners  average_forever  average_2weeks  \
481           0          0       0 .. 20,000                0               0   
7235          9          0  20,000 .. 50,000                0               0   
9553          4          0       0 .. 20,000                0               0   
22244        13          0       0 .. 20,000                0               0   
27324         1          0       0 .. 20,000                0               0   

       median_forever  median_2weeks  price  initialprice  discount  \
481                 0              0    0.0           0.0       0.0   
7235                0              0    0.0           0.0       0.0   
9553                0              0  899.0         899.0       0.0   
22244               0              0   99.0          99.0       0.0   
27324               0              0  199.0         199.0       0.0   

                                                                                                                                                                                                                                                                                        languages  \
481                                                                                                                                                                                                                                                                                           NaN   
7235                                                                                                                                                                                                                                                                                          NaN   
9553   English, French, Italian, German, Spanish - Spain, Arabic, Portuguese - Brazil, Bulgarian, Traditional Chinese, Simplified Chinese, Danish, Finnish, Greek, Hungarian, Japanese, Korean, Dutch, Norwegian, Ukrainian, Polish, Portuguese, Romanian, Russian, Thai, Czech, Turkish, Swedish   
22244                                                                                                                                                                                                                                  English, French, Italian, German, Spanish - Spain, Russian   
27324                                                                                                                                                                                                                              English, German, Danish, Japanese, Russian, Simplified Chinese   

                                  genre  ccu  \
481                                 NaN    0   
7235                                NaN    0   
9553                      Casual, Indie    0   
22244  Action, Adventure, Casual, Indie    0   
27324         Action, Indie, Simulation    0   

                                                                                                                                                                                                                                                                                                                                                 tags  
481                                                                                                                                                                                                                                                  

In [27]:
df_app_list[df_app_list['name'].isnull()]

appid name
481     17760  NaN
7235   396420  NaN
9553   460250  NaN
22244  806160  NaN
27324  965340  NaN

In [28]:
df_steam_app_data[df_steam_app_data['name'].isnull()]

type name  steam_appid  required_age is_free controller_support  dlc  \
7235  game  NaN       396420           0.0    True                NaN  NaN   

                                                                                                                                                                                                                                                                                                                                                                                                                                                           detailed_description  \
7235  Spookeningは3Dの恐怖ゲームで、あなたは毎夜に死んでゴーストとして復活します。<br />\r\n村人を怖がらせます。深夜に彼らの魂の力を取って復活します！<br />\r\n<br />\r\nClifftownは幽霊がいる。無知な村人はspookeningにこの状況の発生を求めたことはないが、あなたも毎夜に死にたくない。「Soularパワー」で復活することができる。<br />\r\nSpookeningでは英雄であるハーマンゴーストとしてゲームする。ハーマンの運命を変更して、別の日で生活するチャンスを与えることができるか？<br />\r\nこの世界のルールを勉強するうちに、生きる過程で重要な「Soularパワー」を取ることができる。このパワーを目標に使って一度の生きるチャンスを得る。<br />\r\nしかし、他の方面から見れば簡単ではない。他の人間はあなたを侵入者として自分の場所を守る。怯えさせすぎると、復讐の幽霊はあなたを狩る。<br />\r\n自分を救って！どこのコストでも？   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                 about_the_game  \
7235  Spookeningは3Dの恐怖ゲームで、あなたは毎夜に死んでゴーストとして復活します。<br />\r\n村人を怖がらせます。深夜に彼らの魂の力を取って復活します！<br />\r\n<br />\r\nClifftownは幽霊がいる。無知な村人はspookeningにこの状況の発生を求めたことはないが、あなたも毎夜に死にたくない。「Soularパワー」で復活することができる。<br />\r\nSpookeningでは英雄であるハーマンゴーストとしてゲームする。ハーマンの運命を変更して、別の日で生活するチャンスを与えることができるか？<br />\r\nこの世界のルールを勉強するうちに、生きる過程で重要な「Soularパワー」を取ることができる。このパワーを目標に使って一度の生きるチャンスを得る。<br />\r\nしかし、他の方面から見れば簡単ではない。他の人間はあなたを侵入者として自分の場所を守る。怯えさせすぎると、復讐の幽霊はあなたを狩る。<br />\r\n自分を救って！どこのコストでも？   

                                            short_description  fullgame  \
7235  Spookeningは3Dの恐怖ゲームで、あなたは毎夜に死んでゴーストとして復活します。 村人を怖がらせます。       NaN   

     supported_languages  \
7235                 NaN   

                                                                   header_image  \
7235  https://steamcdn-a.akamaihd.net/steam/apps/396420/header.jpg?t=1545256825   

     website  \
7235     NaN   

                                                                                                 pc_requirements  \
7235  {'minimum': '<strong>Minimum:</strong><br><ul class="bb_ul"><li><strong>OS:</strong> Windows 7</li></ul>'}   

     mac_requirements linux_requirements legal_notice drm_notice  \
7235               []                 []          NaN        NaN   

     ext_user_account_notice developers publishers demos price_overview  \
7235                     NaN        NaN       ['']   NaN            NaN   

     packages package_groups                                        platforms  \
7235      NaN             []  {'windows': True, 'mac': False, 'linux': False}   

     metacritic reviews categories genres screenshots movies recommendations  \
7235        NaN     NaN        NaN    NaN         NaN    NaN             NaN   

      achievements                                   release_date  \
7235  {'total': 0}  {'coming_soon': False, 'date': '1 Nov, 2016'}   

                  support_info background         content_descriptors  
7235  {'url': '', 'email': ''}        NaN  {'ids': [], 'notes': None}